packages/js/src/core/sign-in.ts

```diff
@@ -34,5 +32,11 @@ export const generateSignInUri = ({
     response_type: responseType,
     prompt,
     scope: withReservedScopes(scopes),
-    resource: resources ?? [],
-  })}`;
+  });
+
+  for (const resource of resources ?? []) {
+    urlSearchParameters.append('resource', resource);
+  }
+
+  return `${authorizationEndpoint}?${urlSearchParameters.toString()}`;
```

Comment:
What if the resources is undefined? The current code logic will always return the endpoint with the `?` appended in the end, regardless of the resources existence.